src/main/java/org/apache/commons/lang3/reflect/MethodUtils.java

```diff
@@ -699,8 +710,10 @@ public static Method getMatchingAccessibleMethod(final Class<?> cls,
             MemberUtils.setAccessibleWorkaround(bestMatch);
         }
 
-        if (bestMatch != null && bestMatch.isVarArgs() && bestMatch.getParameterTypes().length > 0 && parameterTypes.length > 0) {
-            final Class<?>[] methodParameterTypes = bestMatch.getParameterTypes();
+        Class<?>[] bestMatch_getParameterTypes = bestMatch.getParameterTypes();
```

Comment:
This change doesn't preserve semantics - if `bestMatch` is `null`, then this code will throw a `NullPointerException`, but the original code will not. Can you restructure it to both get the benefits of typechecking correctly and preserve the original semantics?